### ISSUE TITLE
No announcement of maxlength

### ISSUE THREAD

**Issue by jkshapiro** [nvaccess/nvda#7910]
### Steps to reproduce:
> 1. Open any web browser.
> 2. Browse to a page that contains an &lt:input&gt; or &lt;textarea&gt; with a `maxlength` attribute.
> 3. Type more characters than will fit.

### Expected behavior:
> NVDA stops announcing characters as they are typed. Maybe it plays an error sound instead.

### Actual behavior:
> NVDA continues to announce characters as they are typed. There is no way for the user to know that they are being discarded.

**Comment by feerrenrut** [comment]
This sounds like a fairly important usability feature for NVDA. The first step will be to see if this attribute is exposed by each of the browsers in some way.
Also providing a mechanism for an NVDA user to know the number of characters remaining, or the current count and limit would be even better.

For web developers looking for a solution to this problem **NOW**, there is interesting discussion on:
- https://github.com/wet-boew/wet-boew/issues/7144
- https://stackoverflow.com/questions/24036568/how-to-make-maxlength-attribute-accessible

**Comment by jkshapiro** [comment]
@dravas-nat Not related to NVDA, but as it happens I have a project at https://github.com/jkshapiro/jquery-accessible-character-countdown that I'm planning to convert to a WET4 plugin at some point this year. Would be happy to collaborate.

**Comment by derekriemer** [comment]
In terms of implementing a count-down, I'd recommend a live region counter which counts down after about 20 characters. I'm not sure a global countdown for all fields with a maxlength is the correct implementation, although playing a ding sound when no more character is entered would certainly be nice.

**Comment by jmuheim** [comment]
Is there no update for that? This problem exists since years already.

**Comment by jmuheim** [comment]
By the way, JAWS doesn't announce it, either.

**Comment by lenzls** [comment]
Just wanted to know if something happened on that issue in the last year.

We were a bit surprised, that screenreaders apparently happily ignore the `maxlength` property.

At least some indication that you've reached the max length would be a start.

**Comment by Kseleny** [comment]
Are there any plans to remediate this issue?  JAWS is working on a fix with tentative plans to deploy with their next release. TY!!

**Comment by seanbudd** [comment]
No plans, but I have triaged the issue so you are welcome to pick up the work @Kseleny

**Comment by jonathan-cavey** [comment]
+1 for visibility. 

We have multiple locations where this accessibility feature would be beneficial to our end users.

**Comment by XLTechie** [comment]
CC @jcsteh

**Comment by Adriani90** [comment]
cc: @SaschaCowley it seems this is highly demanded among accessibility testers, and it is also demanded by users. The progress while typing could be indicated by progress bar tones, and maxlength could be indicated by a tone for every typed character when reaching the max length and the caret doesn't move anymore. Is it a big effort to develop something like this?

**Comment by SaschaCowley** [comment]
I don't see a reason why NVDA should expose the `maxlength` property directly - this is not an affordance that sighted users get natively, so is something that is on authors to provide (and there are some truly hideous implementations of this in the wild).
On the other hand, reporting when the maximum length has been reached and no more text is being entered is something that I think we should have.

**Comment by XLTechie** [comment]
Agreed with @SaschaCowley. There are too many ways this is already indicated by page devs (number of chars used, number of chars remaining, etc.). For NVDA to try to indicate this as well, is likely asking for some very confusing situations.

But indicating (with a ding, for example, as described above), when the end of the field has been reached, does seem like something NVDA should do.

**Comment by joanlambert** [comment]
This is affecting the experience of our exam candidates who use NVDA, which is a growing percentage of candidates. With NVDA making a real effort to go up against JAWS, and with this having been a problem since 2015, it would be terrific if this could bubble up to the top of someone's list.
